src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -375,6 +375,10 @@ public void PatternsReduceIdentically(string pattern1, string pattern2)
         [InlineData("a*a*?", "a*")]
         [InlineData("a*?a*", "a*")]
         [InlineData("a*[^a]*", "a*")]
+        [InlineData("[ab]*[^a]", "(?>[ab]*)[^a]")]
```

Comment:
I'm missing something, isn't the purpose of this change to ensure these two are NOT equivalent (the first is not reduced into the second)?

Reply:
Oh nevermind I was put off by Github seemingly showing these applied to `PatternsReduceIdentically()` rather than `PatternsReduceDifferently()`